### ISSUE TITLE
Mission Planner displays Bad Logging but LOG_BACKEND_TYPE=0

### ISSUE THREAD

**Issue by otsuka-kohei** [ArduPilot/ardupilot#10006]
### Bug report

**Issue details**

When I arm Pixhawk with healthy FAT formatted microSD card but Mission Planner displays "PreArm : Logging Failed" and "Bad Logging".
Therefore I set LOG_BACKEND_TYPE to 0 (disable logging).
But Mission Planner keeps displaying "Bad Logging".
ArduPilot must logging check (microSD check) after check LOG_BACKEND_TYPE parameter.

**Version**
ArduCopter V3.6.3 Quad

**Platform**
[  ] All
[  ] AntennaTracker
[x] Copter
[  ] Plane
[  ] Rover
[  ] Submarine

**Airframe type**
quad X

**Hardware type**
Pixhawk 2.4.8

**Logs**

**Comment by dposea** [comment]
I was going to open an issue on this also. It frsky passthru checks for an open log before arming, which is not good. If logging fails an error message is already generated. Do we need this at all? Passthru works while generating errors every few seconds. I’ll fix it if  removing it is ok.

**Comment by peterbarker** [comment]
On Wed, 12 Dec 2018, Otsuka Kohei wrote:

> When I arm PX4 with healthy FAT formatted microSD card but Mission Planner
> displays "PreArm : Logging Failed" and "Bad Logging".
> Therefore I set LOG_BACKEND_TYPE to 0 (disable logging).
> But Mission Planner keeps displaying "Bad Logging".

Did you reboot after making tht setting?

Are you sure it's "Bad Logging", not "Logging Failed" MP is showing after 
a reboot?  If it is the latter then you need to tell the vehicle it is OK 
to arm if it is not logging by modifying LOG_BITMASK

**Comment by peterbarker** [comment]
... where by `LOG_BITMASK` I mean `ARMING_CHECK`, obviously.

**Comment by peterbarker** [comment]
On Thu, 13 Dec 2018, David Posea wrote:

> I was going to open an issue on this also. It frsky passthru checks for an
> open log before arming, which is not good. If logging fails an error message
> is already generated. Do we need this at all? Passthru works while
> generating errors every few seconds. I’ll fix it if removing it is ok.

Is this actually the same issue as the OP?  He didn't mention FrSky at 
all.

**Comment by otsuka-kohei** [comment]
I rebooted PX4 after update parameters. 
Mission Planner sometimes shows "PreArm : Logging Failed" before "Bad Logging".
And I set LOG_BITMASK to 0 (disabled) but Mission Planner shows these error messages.

I am using SanDisk 4GB speed class 10 (not UHS) microSD, so I will buy UHS1 or faster microSD.

**Comment by peterbarker** [comment]
On Thu, 13 Dec 2018, Otsuka Kohei wrote:

> I rebooted PX4 after update parameters.
> Mission Planner sometimes shows "PreArm : Logging Failed" before "Bad Logging".
> And I set LOG_BITMASK to 0 (disabled) but Mission Planner shows these error messages.

Setting LOG_BITMASK to 0 doesn't really disable anything; we still log 
things even if it is zero.

Are you saying that if you set LOG_BACKEND_TYPE to 0 and reboot you're 
getting "Logging failed" messages?

**Comment by otsuka-kohei** [comment]
Mission Planner shows "PreArm : Logging Failed" with LOG_BACKEND_TYPE=0.

**Comment by peterbarker** [comment]
On Thu, 13 Dec 2018, Otsuka Kohei wrote:

> Mission Planner shows "PreArm : Logging Failed" with LOG_BACKEND_TYPE=0.

OK, that is expected unless you remove logging from the ARMING_CHECK 
bitmask.  Have you done that?

If you haven't done that you're effectively telling the vehicle, "you must 
be logging to arm - but I'm not going to tell you how to log"

**Comment by rmackay9** [comment]
@otsuka-kohei, Sorry for being annoying but a little semantic correction.  "Pixhawk" is hardware, "ArduPilot" is software.  So we reboot the "Pixhawk" after changing a parameter.

**Comment by dposea** [comment]
"Bad logging" only comes from frsky passthru . I ran a find with grep to check all files. I always get "Bad Logging" until I arm. It's checking sensor_status for MAV_SYS_STATUS_LOGGING, and I haven't tracked that down fully yet. Until I checked the source code I thought it was a ChibiOS problem. It was not happening with NuttX. I can open a new issue when I have all the details since this seems to be a different issue.
